config/envoyconfig/protocols.go

```diff
@@ -33,23 +41,25 @@ func buildUpstreamProtocolOptions(endpoints []Endpoint, forceHTTP2 bool) *envoy_
 				},
 			},
 		}
-	}
-
-	// when using TLS use ALPN auto config
-	tlsCount := 0
-	for _, e := range endpoints {
-		if e.transportSocket != nil {
-			tlsCount++
+	case upstreamProtocolAuto:
```

Comment:
Could you add some contextualizing comments to this case? The fallthrough and empty default is uncommon enough where I'm not sure what it does without searching.

Reply:
there are already comments. I don't know what else to add. I will remove the default case.